language-support/hs/bindings/src/DA/Ledger/Services/TransactionService.hs

```diff
@@ -17,7 +25,105 @@ import Network.GRPC.HighLevel.Generated
 import qualified Com.Digitalasset.Ledger.Api.V1.TransactionService as LL
 import qualified Data.Map as Map
 
--- TODO:: all the other RPCs
+getTransactions :: GetTransactionsRequest -> LedgerService (Stream Transaction)
+getTransactions req =
+    makeLedgerService $ \timeout config -> do
+    stream <- newStream
+    _ <- forkIO $
```

Comment:
This kind of pattern can be a bit dangerous since if the thread dies (e.g. throws an exception) you will never notice. Also if the main thread of a Haskell application shuts down, all other threads die immediately without even running finalizers which can be quite confusing. Usually something like `withAsync` or one of the other functions in `async` is a better way to handle this. That said, no need to address this before merging especially since this pattern is already used in other parts, just something to keep in mind for a future PR.

Reply:
Yes I agree. We should switch to `async` as you suggest.